src/main/java/cart/entity/Product.java

```diff
@@ -0,0 +1,57 @@
+package cart.entity;
+
+public class Product {
+    private static final int MAX_NAME_LENGTH = 20;
+    private static final int MAX_PRICE = 100000000;
+    private static final int MIN_PRICE = 0;
+    private static final int UNIT_OF_PRICE = 10;
+
+    private final Long id;
+    private final String name;
+    private final String image;
+    private final int price;
+
+    public Product(String name, String image, int price) {
+        this(null, name, image, price);
+    }
+
+    public Product(Long id, String name, String image, int price) {
+        validate(name, price);
+        this.id = id;
+        this.name = name;
+        this.image = image;
+        this.price = price;
+    }
+
+    private void validate(String name, final int price) {
+        if (name.isBlank() || MAX_NAME_LENGTH < name.length()) {
+            throw new IllegalArgumentException("상품 이름은 1 ~ 20 길이여야 합니다.");
+        }
```

Comment:
아래와 같이 표현하면 이름 최대길이 값이 변경 됐을 때
예외 메시지를 깜빡하고 변경하지 않는 상황을 방지할 수 있을것같아요 😄 

```suggestion
        if (name.isBlank() || MAX_NAME_LENGTH < name.length()) {
            throw new IllegalArgumentException(String.format("상품 이름은 1 ~ %s 길이여야 합니다.", MAX_NAME_LENGTH));
        }
```

Reply:
헉 감사합니다!!